src/client/camera.cpp

```diff
@@ -343,7 +343,11 @@ void Camera::update(LocalPlayer* player, f32 frametime, f32 busytime, f32 tool_r
 		player_position = player->getParent()->getPosition();
 
 	// Smooth the camera movement when the player instantly moves upward due to stepheight
-	if (player_position.Y > old_player_position.Y) {
+	// Disable smoothing if climbing, swimming upwards, or flying, to avoid upwards offset
+	// of player model when seen in 3rd person view.
+	if (player_position.Y > old_player_position.Y && !player->is_climbing &&
+			!player->swimming_vertical &&
+			!(g_settings->getBool("free_move") && m_client->checkLocalPrivilege("fly"))) {
```

Comment:
Instead of this line it'd be better to apply a fix similar to 0ab580810c6ccebfd57497b9565e3e396d250d70 so that `is_climbing` and `swimming_vertical` can never be true in fly mode.

Reply:
Hm it appear the same check is already used elsewhere, not exactly clean but I guess it can be left as-is.

Reply:
This line comment refers to outdated code, see 3rd commit.

Reply:
> Instead of this line it'd be better to apply a fix similar to 0ab5808 so that is_climbing and swimming_vertical can never be true in fly mode.

Maybe later, this would be a complex change that i consider too risky for 5.3.0.